subprojects/core/src/main/java/org/gradle/api/internal/collections/FilteredCollection.java

```diff
@@ -96,6 +99,11 @@ public int estimatedSize() {
         return collection.estimatedSize();
     }
 
+    @Override
+    public MutationGuard getMutationGuard() {
+        return MutationGuards.of(collection);
```

Comment:
why not just `collection.getMutationGuard()`? Every store should have a mutation guard.

Reply:
Changed.